testing/upload_validation.ts

```diff
@@ -979,6 +979,33 @@ function validateFormulas(schema: z.ZodObject<any>) {
         path: ['networkDomains'],
       },
     )
+    .superRefine((data, context) => {
+      if (data.defaultAuthentication && data.defaultAuthentication.type !== AuthenticationType.None) {
+        return;
+      }
+
+      // if the pack has no default authentication, make sure all formulas don't set connection requirements.
+      ((data.formulas || []) as PackFormulaMetadata[]).forEach((formula, i) => {
+        if (formula.connectionRequirement && formula.connectionRequirement !== ConnectionRequirement.None) {
+          context.addIssue({
+            code: z.ZodIssueCode.custom,
+            path: ['formulas', i],
+            message: 'Formula should not set connectionRequirement for no-auth Packs.',
```

Comment:
Nit, we should always be careful with our external-facing terminology, and we don't use the phrase "no-auth Packs". I would just say something like "Formulas cannot set a connectionRequirement when the Pack does not use user authentication."